### ISSUE TITLE
In Reflection.ts propertyMetadata method not returning Object properties when used with Angular-14

### ISSUE THREAD

**Issue by brajeshkr81** [nbeach/ng-unit#113]
I have upgraded my Angular application from 9 to 14, since then test reading mocked child components properties are started failing.
All child components properties are undefined in tests.

**Comment by nbeach** [comment]
I created a branch that to test Angular 14+ and  though I did encounter some module loading issues with the project, I could not replicate any issues with child inputs. Can you provide and example project and test where this issue is occuring?

**Comment by brajeshkr81** [comment]
@nbeach Thanks for your reply.
Below is my contents of tsconfig.json
{
    "compileOnSave": false,
    "compilerOptions": {
        "downlevelIteration": true,
        "importHelpers": true,
        "outDir": "./dist/out-tsc",
        "sourceMap": true,
        "declaration": false,
        "moduleResolution": "node",
        "experimentalDecorators": true,
        "target": "es2020",
        "typeRoots": [
            "node_modules/@types"
        ],
        "lib": [
            "es2017",
            "dom"
        ],
        "module": "es2020",
        "baseUrl": "./",
        "resolveJsonModule": true,
        "skipLibCheck": true,
        "allowSyntheticDefaultImports": true,
        "emitDecoratorMetadata": true
    },
    "angularCompilerOptions": {
        "preserveWhitespaces": true
    }
}

tsconfig-spec.ts:
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/spec",
    "baseUrl": "./",
    "types": [
      "jasmine",
      "node"
    ]
  },
  "files": [
    "test.ts",
    "polyfills.ts"
  ],
  "include": [
    "../src/**/*.spec.ts",
    "../src/**/*.d.ts"
  ]
}

And these are my Dev dependencies of Angular project:

"devDependencies": {
    "@angular-devkit/build-angular": "^14.2.10",
    "@angular/cli": "^14.2.10",
    "@angular/compiler-cli": "^14.2.12",
    "@angular/language-service": "^14.2.12",
    "@types/chai-dom": "0.0.10",
    "@types/jasmine": "~3.8.0",
    "@types/jasminewd2": "~2.0.10",
    "@types/lodash": "^4.14.109",
    "@types/node": "^12.20.55",
    "@types/react": "^17.0.15",
    "@types/sinon": "^10.0.13",
    "@types/sinon-chai": "^3.2.5",
    "@types/webpack": "~4.1.7",
    "@types/webpack-env": "~1.13.6",
    "chai": "~4.3.4",
    "chai-dom": "~1.9.0",
    "codelyzer": "^6.0.0",
    "husky": "^1.3.1",
    "jasmine-core": "~3.8.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "^6.4.2",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-firefox-launcher": "~2.1.1",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "karma-sauce-launcher": "^4.1.4",
    "karma-spec-reporter": "0.0.32",
    "less": "^3.12.2",
    "ng-unit": "~1.1.1",
    "sinon": "~14.0.0",
    "sinon-chai": "~3.7.0",
    "stringstream": "~0.0.6",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "tslint-no-unused-expression-chai": "^0.1.4",
    "tunnel-agent": "~0.6.0",
    "typescript": "~4.6.4",
    "url-parse": "^1.5.4"
  }

When i debugged my test, i found that in "propertyMetadata" method of "reflection.js" from ng-unit library 
line ----> return (object as any)**.__prop__metadata_**_ ||  Reflect.getMetadata("propMetadata", object) || {}

My Component class i.e. object in above code, don't have any property  **_prop_metadata**. Instead of that it has **propDecorators** property.

So i tried overriding "propertyMetadata" method of reflection.js in my "test.ts" like below. And luckily able to fetch all child mock components  properties.

const reflection = require('../node_modules/ng-unit/src/reflection.js')
delete reflection['propertyMetadata']
reflection.propertyMetadata = function propertyMetadata(object) {
    let props = cloneDeep(object.propDecorators);
    if(props) {
        Object.keys(props).forEach(key => {
            if (typeof props[key][0]?.type == 'function') {
                props[key][0] = new props[key][0].type();
            }
        })
        return props;
    }
    return Reflect.getMetadata("propMetadata", object) || {};
}


I am not sure, if this is due to update typeScript version, or the ES version i am using in tsconfig.

If you still not able to understand the problem, please reply back.  i will create a small project and share.

**Comment by nbeach** [comment]
I can't apply that config the the project without seeing other issues. Can generate a new angular app with the angular cli, replicate the issue you're seeing in that, and link me to a repo containing that?

**Comment by brajeshkr81** [comment]
@nbeach I have create a small test project https://github.com/brajeshkr81/test-ng-unit, please checkout and run test for parent component.

I am also attaching the snap of debug. which points where the issue is.
<img width="960" alt="debug_screenshot" src="https://github.com/nbeach/ng-unit/assets/33995599/871d5d4f-8497-47c5-92f7-eb05c77edfb9">

**Comment by nbeach** [comment]
Thank you for this! Once I was able to reproduce this with confidence I made a couple changes to how we detect decorator metadata, one being the one you pointed out, and released v1.2.1. With this in place all tests in that project now pass. Though I did have to add a `detectChanges()` to `src/app/parent/parent.component.spec.ts` in order for it to pass, but this was to be expected. 

Let me know if this resolves your issue.
